File: Java/src/main/java/Solution501_600/Solution575.java
package Solution501_600;

import java.util.HashSet;

/**
 * @author windows
 */
public class Solution575 {
    public int distributeCandies(int[] candyType) {
        HashSet<Integer> set = new HashSet<>();
        for (int i = 0; i < candyType.length; i++) {
            set.add(candyType[i]);
        }
        return Math.min(set.size(), candyType.length /2);
    }



    public static void main(String[] args) {
        Solution575 solution575 = new Solution575();
    }
}

File: C++/src/Solution501_600/Solution575.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution575.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <unordered_set>

int Solution575::distributeCandies(vector<int> &candyType) {
    int n = candyType.size();
    unordered_set<int> hashset;
    for (int i = 0; i < n; ++i) {
        hashset.insert(candyType[i]);
    }
    return hashset.size() >= n / 2 ? n / 2: hashset.size();
}
